### PR TITLE
Testing Logic and Components

### PR DESCRIPTION
**Pull request by balerum03**
Added test:
- Unit testing for Logic.
- Snapshot testing for React components.

### CODE REVIEW

**Review by uche-inyama — CHANGES_REQUESTED** [review]
## Changes Required

### Hi balerum03 :slightly_smiling_face: 

Good job, you are almost through with the tests. It is required that all functions in the calculate.js file are covered with unit tests, but tests for **%** and **+/-** were not covered.

#### Make these updates then request for another code review.

### Happy coding :man_technologist: 

#### ``` Uchechukwu Henry Inyama ```

File: .env [hunk 1]
```diff
@@ -0,0 +1 @@
+SKIP_PREFLIGHT_CHECK=true
```
Comment (Forison) [inline .env]: * [x] Never commit .env file to repo, add the path to your `gitignore` so they don't get committed to the remote branch

File: src/logic/__test__/calculate.test.js [hunk 2]
```diff
@@ -0,0 +1,70 @@
...
+describe('User can use +/-', () => {
+  test('user changes the from negative to positive', () => {
+    const result = calculate({
+      total: '', next: '1', operation: '',
+    }, '+/-');
+    expect(result.total.s).toEqual(-1);
+  });
+});
+
+describe('User can user %', () => {
```
Comment (Forison) [inline src/logic/__test__/calculate.test.js]: * [x] Let us improve the description here, I bet you wanted to say `User can use %`. Let us rectify this.

File: src/logic/__test__/calculate.test.js [hunk 3]
```diff
@@ -0,0 +1,70 @@
...
+    }, '2');
+    expect(result.next).toEqual('2');
+    result = calculate(result, '.');
+    result = calculate(result, '3');
+    expect(result.next).toEqual('2.3');
+  });
+});
+
+describe('User can use +/-', () => {
+  test('user changes the from negative to positive', () => {
```
Comment (Forison) [inline src/logic/__test__/calculate.test.js:55]: * [x] Let us add a scenario where ie user enters `2` `+` `4` and then selects `+/-`

**Review by Forison — CHANGES_REQUESTED** [review]
## Status::Changes Required ❌

# 👋 @balerum03 

#### Good job on adding test to cover `%` and `+/-` as requested earlier, however there are few issues we have to address to the project approved😇.

* [x] Kindly check my comments below👇.

#### That is it, kindly make the changes and submit a re-review💪

#### Happy coding👨‍💻

Addo Forison

File: src/logic/__test__/calculate.test.js [hunk 4]
```diff
@@ -0,0 +1,70 @@
...
+});
+
+describe('User can user %', () => {
+  test('user clicks on % after a number', () => {
+    const result = calculate({
+      total: '', next: '100', operation: '',
+    }, '%');
+    expect(result.total.c).toEqual([1]);
+  });
+});
```
Comment (Forison) [inline src/logic/__test__/calculate.test.js:81]: * [ ] Let us add a scenario where ie user enters `2` `+` `4` and then selects `%`

**Review by mikenath223 — APPROVED** [review]
### CONGRATULATIONS  👏👏👏

This project meets the requirements for approval.

**STATUS: APPROVED ✅**


Don't forget to fill the project completion form👍
Good luck with the next project.

**HAPPY CODING 🎉💻🎉**

**Michgolden Ukeje (CR)**

**Pull request merged**